SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,501 @@
+				   TEXT("EntityId: %lld Delegating skeleton entity to VirtualWorker %d Partition %lld"), CreatedSkeletonEntityId,
+				   AuthWorkerId, AuthWorkerPartition);
+
+			AuthorityDelegation AuthorityDelegationComponent;
+			AuthorityDelegationComponent.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthWorkerPartition);
+
+			EntitiesToPopulatingWorkers.Emplace(CreatedSkeletonEntityId, AuthWorkerId);
+
+			NetDriver->Connection->GetCoordinator().SendComponentUpdate(
+				CreatedSkeletonEntityId,
+				ComponentUpdate(OwningComponentUpdatePtr(AuthorityDelegationComponent.CreateAuthorityDelegationUpdate().schema_type),
+								AuthorityDelegation::ComponentId),
+				{});
+		}
+
+		SkeletonEntitiesToDelegate.Empty();
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Finished delegating entities, moving on to manifest"));
+
+		Stage = EStage::SigningManifest;
+	}
+
+	if (Stage == EStage::SigningManifest)
+	{
+		TMap<VirtualWorkerId, TArray<Worker_EntityId>> WorkerToEntitiesToPopulate;
+		// Every worker should get a manifest to proceed through the skeleton entity flow.
+		for (const auto& VirtualWorker : NetDriver->WellKnownEntitySystem->VirtualWorkerTranslationManager->GetVirtualWorkerMapping())
+		{
+			WorkerToEntitiesToPopulate.FindOrAdd(VirtualWorker.Key);
+		}
+		for (const auto& SkeletonEntityToPopulatingWorker : EntitiesToPopulatingWorkers)
+		{
+			WorkerToEntitiesToPopulate.FindOrAdd(SkeletonEntityToPopulatingWorker.Value).Add(SkeletonEntityToPopulatingWorker.Key);
+		}
+		for (const auto& WorkerToEntitiesToPopulateEl : WorkerToEntitiesToPopulate)
+		{
+			const VirtualWorkerId WorkerId = WorkerToEntitiesToPopulateEl.Key;
+			const TArray<Worker_EntityId>& EntitiesToPopulate = WorkerToEntitiesToPopulateEl.Value;
+
+			TArray<ComponentData> SkeletonEntityManifestComponents;
+
+			{
+				FSkeletonEntityManifest Manifest;
+				Manifest.EntitiesToPopulate = EntitiesToPopulate;
+
+				SkeletonEntityManifestComponents.Emplace(Manifest.CreateComponentData());
+			}
+
+			const Worker_PartitionId ServerWorkerPartitionId =
+				NetDriver->VirtualWorkerTranslator->GetPartitionEntityForVirtualWorker(WorkerId);
+
+			{
+				const Worker_PartitionId ThisWorkerPartitionId = NetDriver->VirtualWorkerTranslator->GetClaimedPartitionId();
+				AuthorityDelegation Authority;
+				Authority.Delegations.Emplace(SpatialConstants::SKELETON_ENTITY_MANIFEST_AUTH_COMPONENT_SET_ID, ServerWorkerPartitionId);
+				Authority.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, ThisWorkerPartitionId);
+
+				SkeletonEntityManifestComponents.Emplace(
+					ComponentData(OwningComponentDataPtr(Authority.CreateComponentData().schema_type), AuthorityDelegation::ComponentId));
+			}
+
+			{
+				Query Q;
+				Q.Constraint.bSelfConstraint = true;
+				Q.ResultComponentIds.Append(
+					{ SpatialConstants::SKELETON_ENTITY_MANIFEST_COMPONENT_ID, SpatialConstants::AUTHORITY_DELEGATION_COMPONENT_ID });
```

Comment:
Why is AUTHORITY_DELEGATION_COMPONENT_ID needed ?

Reply:
Not needed anymore, thank you - did updates to it in a previous version.